README.md

```diff
@@ -33,57 +41,252 @@ TBD
-cd ezcap-client
+git clone https://github.com/digitalbazaar/ezcap.git
+cd ezcap
 npm install
 ```
 
 ## Usage
 
 * [Creating a Client](#creating-a-client)
-* [Invoking a Capability](#invoking-a-capability)
+* [Reading with a Root Capability](#reading-with-a-root-capability)
+* [Writing with a Root Capability](#writing-with-a-root-capability)
 * [Delegating a Capability](#delegating-a-capability)
+* [Reading with a Delegated Capability](#reading-with-a-delegated-capability)
+* [Writing with a Delegated Capability](#writing-with-a-delegated-capability)
+* [Invoking with a Root Capability](#invoking-a-capability)
+* [Invoking with a Delegated Capability](#invoking-a-capability)
 
 ### Creating a Client
 
 ```js
-import {Ed25519KeyPair} from 'crypto-ld';
-import {EzcapClient} from 'ezcap-client';
+import {ZcapClient} from 'ezcap';
 import didKey from 'did-method-key';
-const {keyToDidDoc} = didKey.driver();
+const didKeyDriver = didKey.driver();
 
-// generate a did:key to use with the ezcap client
-const keypair = await Ed25519KeyPair.generate();
-const {id} = await keyToDidDoc(keypair);
+// the base URL for the zcap client to operate against
+const baseUrl = 'https://zcap.example';
 
-const ezcap = new EzcapClient({id, keypair});
+// generate a DID Document and set of key pairs
+const {didDocument, keyPairs} = await didKeyDriver.generate();
+
+// create a new zcap client using the generated cryptographic material
+const zcapClient = new ZcapClient({baseUrl, didDocument, keyPairs});
 ```
 
-### Invoking a Capability
+### Reading with a Root Capability
 
 ```js
-const method = 'POST';
-const capability = 'https://zcap.example/foo';
-const payload = {foo: 'bar'};
-const response = await ezcap.invoke({method, capability, payload});
+const url = '/my-account/items';
+
+// reading a URL using a zcap will result in an HTTP Response
+const response = await zcapClient.read({url});
+
+// retrieve the JSON data
+const items = await response.json();
+```
+
+### Writing with a Root Capability
+
+```js
+const url = '/my-account/items';
+const item = {label: 'Widget'};
+
+// writing a URL using a zcap will result in an HTTP Response
+const response = await zcapClient.write({url, json: item});
+
+// process the response appropriately
+const writtenItem = await response.json();
 ```
 
 ### Delegating a Capability
 
 ```js
 const capability = 'https://zcap.example/foo';
-const delegate = 'did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH';
-const actions = ['read'];
-const delegatedCapability = ezcap.delegate({capability, delegate, actions});
+const targetDelegate =
+  'did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH';
+const allowedActions = ['read'];
+const delegatedCapability = zcapClient.delegate({capability, targetDelegate, allowedActions});
 ```
 
-## Contribute
+### Reading with a Delegated Capability
+
+```js
+const url = '/my-account/items/123';
+const capability = await getCapabilityFromDatabase({url}); // defined by your code
```

Comment:
I'm not sure that this will be how capabilities are commonly retrieved; e.g., we often use a reference ID to find zcaps in our applications.